### DISCUSSION TITLE
How to use Envelop for server-side rendering

### DISCUSSION THREAD

**Discussion by vladinator1000** [dotansimha/envelop#2250] [Q&A]
Hey folks, I have [this Remix app](https://github.com/vladinator1000/remix-apollo-ssr-pagination/blob/main/app/graphql/apollo.server.ts) that uses Apollo client with SchemaLink to avoid the network. It's running in Cloudflare Pages. 

I'm creating my client for server rendering like this
```tsx
export function createGraphqlClient(context: Dependencies) {
  return new ApolloClient({
    cache: createCache(),
    ssrMode: true,
    link: ApolloLink.from([
      // logLink,
      new SchemaLink({ schema: context.gqlSchema, context }),
    ]),
  })
}
```

How do you set up envelop with Apollo client (not server)? I can't figure it out. My goal is to use the same middleware for both server rendering and `/graphql` endpoints. Things like tracing, query authorization, etc.

I did find this [workers API example](https://github.com/n1ru4l/envelop/blob/main/examples/cloudflare-workers/index.ts), but I couln't find anything on server-side rendering.

I read the [SchemaLink source](https://github.com/apollographql/apollo-client/blob/main/src/link/schema/index.ts#L46C1-L49C40) and it looks like it doesn't include a way to override `execute` and `validate`. I could write my own "EnvelopSchemaLink" based on that but I'm curious how everyone else handles server-side rendering. I'm porting a large Apollo app so I'm stuck with it for better or worse.

**Comment by EmrysMyrddin** [comment]
Hi! Thank you for reaching out!

Are you using Graphql Yoga for your server ?

**Reply by vladinator1000** [reply]
Hi @EmrysMyrddin, I'm not using Yoga for this, my app is a Remix monolith that does server-side rendering using Apollo with SchemaLink to avoid the network. You could say that SchemaLink is my "server". I created an example with a possible implementation for it here: https://github.com/n1ru4l/envelop/issues/2256#issue-2367930548
